File: src/dependent/notifications/TemplateRenderer.java
package org.example.dependent.notifications;

import java.util.Map;

public class TemplateRenderer {
    public String render(String templateName, Map<String, Object> data) {
        StringBuilder builder = new StringBuilder();
        builder.append("TEMPLATE[").append(templateName).append("]:");
        for (Map.Entry<String, Object> entry : data.entrySet()) {
            builder.append(entry.getKey()).append("=").append(String.valueOf(entry.getValue())).append(";");
        }
        return builder.toString();
    }
}

File: src/dependent/notifications/template_renderer.py
from typing import Any, Dict


class TemplateRenderer:
    def render(self, template_name: str, data: Dict[str, Any]) -> str:
        parts = [f"TEMPLATE[{template_name}]:"]
        for key, value in data.items():
            parts.append(f"{key}={value};")
        return "".join(parts)
